examples/inspection/plot_causal_interpretation.py

```diff
@@ -0,0 +1,152 @@
+"""
+===================================================
+Failure of Machine Learning to infer causal effects
+===================================================
+
+Machine Learning models are great for measuring statistical associations.
+Unfortunately, unless we're willing to make strong assumptions about the data,
+those models are unable to infer causal effects.
+
+To illustrate this, we will simulate a situation in which we try to answer
+one of the most important questions in economics of education:
+what is the causal effect of earning a college degree on hourly wages?
+Although the answer to this question is crucial to policy makers,
+`Omitted-Variable Biases <https://en.wikipedia.org/wiki/Omitted-variable_bias>`_ (OVB)
+prevent us from identifying that causal effect.
+
+
+.. contents::
+   :local:
+   :depth: 1
+"""
+
+print(__doc__)
+
+# %%
+# The dataset: simulated hourly wages
+# -----------------------------------
+#
+# The data generating process is laid out in the code below.
+# Work experience in years and a measure of ability are drawn
+# from a Normal distribution;
+# the hourly wage of one of the parents is drawn from Beta distribution.
+# We then create an indicator of college degree which is positivily
+# impacted by ability and parental hourly wage. Finally, we model
+# hourly wages as a linear function of all the previous variables and a
+# random component. Note that all variables have a positivie effect on
+# hourly wages.
+
+import numpy as np
+
+n_samples = 10000
+rng = np.random.RandomState(32)
+
+experiences = rng.normal(20, 10, size=n_samples).astype(int)
+experiences[experiences < 0] = 0
+abilities = rng.normal(0, 0.15, size=n_samples)
+parent_hourly_wages = 50 * rng.beta(2, 8, size=n_samples)
+parent_hourly_wages[parent_hourly_wages < 0] = 0
+
+college_degrees = (9 * abilities + 0.02 * parent_hourly_wages
+                   + rng.randn(n_samples) > 0.7).astype(int)
+
+hourly_wages = (
+  0.2 * experiences + parent_hourly_wages + 2
+  * college_degrees + 5 * abilities + rng.normal(0, 1, size=n_samples)
+)
+
+hourly_wages[hourly_wages < 0] = 0
+
+# %%
+# Description of the simulated data
+# ---------------------------------
+#
+# The following plot shows the distribution of each variable,
+# and pairwise scatter plots. Key to our OVB story is the positive
+# relationship between ability and college degree.
+
+import pandas as pd
+import seaborn as sns
+
+df = pd.DataFrame({
+    'hourly wage': hourly_wages,
+    'experience': experiences,
+    'parent hourly wage': parent_hourly_wages,
+    'college degree': college_degrees,
+    'ability': abilities
+    })
+
+_ = sns.pairplot(df, diag_kind='kde')
```

Comment:
What do you think about changing the order of `df` and adding `corner=True` to the `pairplot`?

```python
import pandas as pd
import seaborn as sns

df = pd.DataFrame({
    'college degree': college_degrees,
    'ability': abilities,
    'hourly wage': hourly_wages,
    'experience': experiences,
    'parent hourly wage': parent_hourly_wages,
    })

grid = sns.pairplot(df, diag_kind="kde", corner=True)
```

This way the relationship between ability and college degree is on the upper left:

<img width="817" alt="Screen Shot 2021-10-05 at 5 46 08 PM" src="https://user-images.githubusercontent.com/5402633/136107266-4bc9d48a-f1ef-441e-b506-ca3596f6984d.png">